tensorboard/webapp/widgets/line_chart_v2/sub_view/line_chart_axis_view.ts

```diff
@@ -59,40 +69,62 @@ export class LineChartAxisComponent {
 
   editMenuOpened = false;
 
-  setEditMenuOpened(opened: boolean): void {
-    this.editMenuOpened = opened;
+  majorTicks: MajorTick[] = [];
+  minorTicks: MinorTick[] = [];
+
+  ngOnChanges() {
+    const {major, minor} = this.getMajorMinorTicks();
+    this.majorTicks = major;
+    this.minorTicks = minor;
   }
 
   getFormatter(): Formatter {
     return this.customFormatter ?? this.scale.defaultFormatter;
   }
 
-  /**
-   * Returns true if the major ticks must be shown.
-   *
-   * Major tick is required for temporal axis since tick string do not have sufficient
-   * information alone and is hard to fit time information without a major ticks.
-   */
-  shouldShowMajorTicks(): boolean {
-    const majorTickCount = this.getMajorTicks().length;
-    const minorTickCount = this.getTicks().length;
+  trackByMinorTick(tick: MinorTick): number {
+    return tick.value;
+  }
 
-    return (
-      this.scale instanceof TemporalScale &&
-      this.axisExtent[1] - this.axisExtent[0] < DAY_IN_MS &&
-      minorTickCount > majorTickCount &&
-      majorTickCount <= 2
-    );
+  trackByMajorTick(tick: MajorTick): number {
+    return tick.start;
   }
 
-  getMajorTicks(): number[] {
-    return this.scale.ticks(this.axisExtent, 2);
+  private getMajorTickValues(): number[] {
+    if (this.scale instanceof TemporalScale) {
+      const majorTicks = this.scale.ticks(this.axisExtent, 2);
+      if (
+        this.axisExtent[1] - this.axisExtent[0] < DAY_IN_MS &&
+        majorTicks.length <= 2
+      ) {
+        return majorTicks;
+      }
+    }
+
+    return [];
   }
 
-  getTicks(): number[] {
+  private getMajorMinorTicks(): {major: MajorTick[]; minor: MinorTick[]} {
+    const formatter = this.getFormatter();
     const domSize = this.axis === 'x' ? this.domDim.width : this.domDim.height;
     const maxTickSize = getDomSizeInformedTickCount(domSize, this.gridCount);
-    return this.scale.ticks(this.axisExtent, maxTickSize);
+    const minorTicks: MinorTick[] = [];
+    for (const tickValue of this.scale.ticks(this.axisExtent, maxTickSize)) {
+      minorTicks.push({
+        tickFormattedString: formatter.formatTick(tickValue),
+        value: tickValue,
```

Comment:
nit: I prefer the same order as the interface defines (alphabetically)

Reply:
This is alphabetical as is and I do not think keeping every object defined in the order of definition in an interface is tenable/more-readable/enforceable. Will keep it as is.